vllm/sampling_params.py

```diff
@@ -153,6 +182,12 @@ class SamplingParams(
     output_text_buffer_length: int = 0
     _all_stop_token_ids: Set[int] = msgspec.field(default_factory=set)
 
+    # Fields used to construct logits processors
+    guided_decoding: Optional[GuidedDecodingParams] = None
+    logit_bias: Optional[Union[Dict[int, float], Dict[str, float]]] = None
+    allowed_token_ids: Optional[List[int]] = None
```

Comment:
This is the bit I want feedback on- how should these params look?

There were enough parameters related to just guided decoding that it seemed to make sense to nest them into one object. The goal here would be to delete the values in these fields and populate the `logits_processors` field by the time the request makes it to a model, so I'm assuming that using a `@dataclass` for simplicity won't cause any performance issues if it never makes it down the stack past the engine layer.

Reply:
cc @rkooo567 since you worked on #7109